### PR TITLE
Use google cloud collector exporter, and feature-gate pdata-direct metrics exporter.

### LINKED ISSUE #7132: [exporter/googlecloud] Re-write exporter to remove opencensus translation and add integration testing
**Current state**

The current googlecloud exporter converts OTLP metrics to opencensus, before sending them using the [opencensus stackdriver exporter](https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver).  The exporter only has unit tests.

**Requirements**

* The exporter does not rely on opencensus libraries or APIs, except for self-observability metrics.
* The exporter is tested against Google Cloud Monitoring's APIs during presubmits.
* The exporter does not rely on (unstable) opentelemetry APIs, such as the (unstable) [googlecloud metrics exporter](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/tree/main/exporter/metric).
* The new exporter fully supports the existing exporter configurations
* The new exporter supports all existing self-observability metrics
* During a migration period, users can switch back to the opencensus-based exporter with a flag

**Proposal**

In order to have integration tests against GCM apis, most of the exporter will live in the [github.com/GoogleCloudPlatform/opentelemetry-operations-go](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go) repo.  Presubmits will be run against PRs to that repo.  The collector exporter will not depend on the go SDK metrics exporter.  The new, re-written exporter will be gated by a [feature-gate](github.com/open-telemetry/opentelemetry-collector/blob/main/service/featuregate/README.md#collector-feature-gates).

This means the googlecloud exporter will have a dependency on [github.com/GoogleCloudPlatform/opentelemetry-operations-go](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go), which will have a dependency on https://github.com/open-telemetry/opentelemetry-collector.

**History**

Github discussion (8/4/21): https://github.com/open-telemetry/opentelemetry-collector-contrib/discussions/4439
Raised at the [8/11/21 collector sig meeting](https://docs.google.com/document/d/1r2JC5MB7GupCE7N32EwGEXs9V_YIsPgoFiLP4VWVMkE/edit#heading=h.fbtfpmbdlf0i)

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7132

This changes the googlecloud exporter to use the newly developed [GCP collector exporter](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/tree/col-exporter-rewrite/exporter/collector) in the github.com/GoogleCloudPlatform/opentelemetry-operations-go repository.  The googlecloud collector exporter lives in that repository so that e2e tests can be run against live google cloud APIs at presubmit time.

Changes are gated by the `exporter.googlecloud.OTLPDirect` feature-gate.  This feature will initially be alpha while we iron out any bugs we find.

### Factory changes and feature-gating

The feature-gate applies to the factory so that the feature-gate also changes which config structure is valid.  There are [breaking changes across client options](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/breaking-changes.md#configuration).  The feature gate also changes which implementation of the metrics and traces exporter is used.

### Trace exporter

The only difference between the new vs old trace exporter is configuration (see feature-gate section above); it is unchanged otherwise.

### Metric exporter

The metrics exporter is completely re-written, and converts directly from pdata to google cloud's types.  Since this is a new implementation, leave the previous implementation in-place in this repository, but prefix files with legacy as a reminder that they should be removed when the feature is graduated to GA.  The new metrics implementation is only used when the feature gate is enabled.

### Dependencies

The collector exporter is designed specifically to minimize its dependencies on collector-core.  It does this by _not_ implementing the component interfaces (e.g. `NewFactory()`), and does not depend on component types (e.g. `component.ExporterCreateSettings`).  Thus, it must define its own `Config`, which is the portion of the overall `Config` which does not include component types.  The remaining code here opentelemetry-collector-contrib is just a wrapper around the `Config` of the GCP collector exporter which adds exporterhelper pieces, and implements the required collector interfaces.

The only dependencies the GCP collector exporter has on collector core are: 
* `go.opentelemetry.io/collector/model/pdata`
* `go.opentelemetry.io/collector/model/semconv/v1.8.0`

### CODE REVIEW

**Comment by dashpole** [comment]
This is ready for review.

**Review by jsuereth — APPROVED** [review]

File: exporter/googlecloudexporter/README.md [hunk 1]
```diff
@@ -204,6 +205,13 @@ and [memory limiter](https://github.com/open-telemetry/opentelemetry-collector/t
 optimal network usage and avoiding memory overruns.  You may also want to run an additional
 [sampler](../../processor/probabilisticsamplerprocessor), depending on your needs.
 
+## Features and Feature-Gates
+
+See the [Collector feature gates](https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/featuregate/README.md#collector-feature-gates) for an overview of feature gates in the collector.
+
+**ALPHA**: `exporter.googlecloud.OTLPDirect`
+
+When enabled via `--feature-gates=exporter.googlecloud.OTLPDirect`, the googlecloud exporter translates pdata directly to google cloud monitoring's types, rather than first translating to opencensus.  See the [Breaking Changes documentation](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/breaking-changes.md#breaking-changes-vs-old-googlecloud-exporter) for breaking changes that will occur as a result of enabling this feature.
```
[the remark this answers is not in the document]
Comment (dashpole) [reply exporter/googlecloudexporter/README.md:214]: I added a section documenting the new config.

**Comment by dashpole** [comment]
@djaglowski @jpkrohling let me know if there is anything I can do to help make reviews easier.

**Comment by jpkrohling** [comment]
@open-telemetry/collector-contrib-approvers, could one of you review this one? I don't think I can review this over the next few days.

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
There are a bunch of conflicts which you should fix, but otherwise LGTM

File: exporter/googlecloudexporter/README.md [hunk 2]
```diff
@@ -58,6 +58,7 @@ These instructions are to get you up and running quickly with the GCP exporter i
           exporters: [googlecloud, logging]
         metrics:
           receivers: [otlp]
+          # TODO(dashpole): remove batching when OTLPDirect goes to beta, since the exporter already does it.
```
[the remark this answers is not in the document]
Comment (dashpole) [reply exporter/googlecloudexporter/README.md]: Removed the TODO, and added a bullet point below talking about the change in exporter behavior.  When it goes beta, I'll invert the docs (have a section below for what happens when the feature gate is disabled).

**Pull request merged**